src/sources/journald.rs

```diff
@@ -643,19 +648,25 @@ impl Finalizer {
         acknowledgements: bool,
         checkpointer: SharedCheckpointer,
         shutdown: ShutdownSignal,
-    ) -> Self {
+    ) -> (Self, BoxFuture<'static, ()>) {
         if acknowledgements {
-            let (finalizer, mut ack_stream) = OrderedFinalizer::new(shutdown);
+            let (finalizer, mut ack_stream) = OrderedFinalizer::new(shutdown.clone());
+            let (trigger, tripwire) = oneshot::channel();
+            let mut trigger = Some(trigger);
             tokio::spawn(async move {
                 while let Some((status, cursor)) = ack_stream.next().await {
-                    if status == BatchStatus::Delivered {
+                    if trigger.is_some() && status == BatchStatus::Delivered {
                         checkpointer.lock().await.set(cursor).await;
+                    } else if let Some(trigger) = trigger.take() {
```

Comment:
Not sure I fully understand this. Rather than wrapping `trigger` in an `Option`, is there a reason why we aren't just breaking out of the loop on an error?

If we have a batch that errors, followed by a batch that succeeds would this mean the checkpoint is set to the end of the successful batch and the errored batch is ignored in a future run? 

Or will the `trigger.send` mean that the `ack_stream` terminates at that point anyway, so we don't get the ensuing successful batch? In that case isn't that the same think as breaking out of the loop?

Reply:
Addressed in fb25b996b